cranelift-reader/src/parser.rs

```diff
@@ -850,33 +858,47 @@ impl<'a> Parser<'a> {
         }
     }
 
-    /// Parse a list of literals (i.e. integers, floats, booleans); e.g.
-    fn parse_literals_to_uimm128(&mut self, ty: Type) -> ParseResult<V128Imm> {
+    /// Parse a list of literals (i.e. integers, floats, booleans); e.g. `0 1 2 3`, usually as
+    /// part of something like `vconst.i32x4 [0 1 2 3]`.
+    fn parse_literals_to_constant_data(&mut self, ty: Type) -> ParseResult<ConstantData> {
         macro_rules! consume {
             ( $ty:ident, $match_fn:expr ) => {{
                 assert!($ty.is_vector());
-                let mut v = Vec::with_capacity($ty.lane_count() as usize);
+                let mut data = ConstantData::default();
                 for _ in 0..$ty.lane_count() {
-                    v.push($match_fn?);
+                    data = data.append($match_fn);
                 }
-                V128Imm::from_iter(v)
+                data
             }};
         }
 
+        fn boolean_to_vec(value: bool, ty: Type) -> Vec<u8> {
+            let lane_size = ty.bytes() / u32::from(ty.lane_count());
+            if lane_size < 1 {
+                panic!("SizedBoolean must have a byte size greater than zero.");
```

Comment:
nit: SizedBoolean isn't a thing anymore.

(Thanks for implementing it this way! I like a lot how it's self-contained.)